### PR TITLE
Fix non-standard syntax

### PR DESCRIPTION
**Pull request by tnajdek**

### CODE REVIEW

File: chrome/content/zotero/charsetMenu.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ var Zotero_Charset_Menu = new function() {
 			Components.utils.import("resource://gre/modules/CharsetMenu.jsm");
 			var cmData = CharsetMenu.getData();
 			for (let charsetList of [cmData.pinnedCharsets, cmData.otherCharsets]) {
-				for each(var charsetInfo in charsetList) {
+				for(let charsetInfo in charsetList) {
```
Comment (dstillman) [inline chrome/content/zotero/charsetMenu.js]: Should be `of`, no?

(Also, space after `for`.)

**Comment by tnajdek** [comment]
good catch! I've updated the commit to fix this.

File: chrome/content/zotero/preferences/preferences_sync.js [hunk 2]
```diff
@@ -639,7 +639,7 @@ Zotero_Preferences.Sync = {
 						// TODO: better error handling
 						
 						// Verify username and password
-						var callback = function () {
+						var callback = function* () {
```
Comment (adomasven) [inline chrome/content/zotero/preferences/preferences_sync.js]: `Zotero.Promise.coroutine(function()* {` here

**Comment by tnajdek** [comment]
Tweaked as per @adomasven comment above

**Pull request merged**